react/features/base/avatar/components/AbstractAvatar.js

```diff
@@ -0,0 +1,215 @@
+// @flow
+
+import { getGravatarURL } from 'js-utils/avatar';
+import { PureComponent } from 'react';
+
+import { getAvatarColor, getInitials } from '../functions';
+
+import { getParticipantById } from '../../participants';
+
+export type Props = {
+
+    /**
+     * URI of the avatar, if identifiable.
+     */
+    _avatarURL: ?string,
+
+    /**
+     * URI of the gravatar if identifyable.
+     */
+    _gravatarURL: ?string,
+
+    /**
+     * The string we base the initials on (this is generated from a list of precendences).
+     */
+    _initialsBase: ?string,
+
+    /**
+     * A string to override the initials to generate a color of. This is handy if you don't want to make
+     * the background color match the string that the initials are generated from.
+     */
+    colorBase?: string,
+
+    /**
+     * Display name of the entity to render an avatar for (if any). This is handy when we need
+     * an avatar for a non-participasnt entity (e.g. a recent list item).
+     */
+    displayName?: string,
+
+    /**
+     * The ID of the participant to render an avatar for (if it's a participant avatar).
+     */
+    participantId?: string,
+
+    /**
+     * The size of the avatar.
+     */
+    size: number,
+
+    /**
+     * URI of the avatar, if any.
+     */
+    uri: ?string,
```

Comment:
This and _avatarURL seem to mean the same thing. Can we perhaps clarify by using some other name or coalesce both? Also, URI vs URL, let's be consistent.

Reply:
This got fully refactored, thanks.

Though I'll keep an eye on url vs uri. URI is my pick though as it can also identify other resources as well